bin/mmdb_utils.py

```diff
@@ -129,24 +138,47 @@ def __init__(self, session_key):
         if not os.path.exists(LOOKUP_DIR_LOCATION):
             os.makedirs(LOOKUP_DIR_LOCATION)
 
-        account_id = self.get_max_mind_account_id()
-        if not account_id:
+        mmdb_config = self.get_max_mind_config()
+        if not mmdb_config:
             msg = "Max Mind Account key not found. Please update config from Max Mind database configuration page."
             logger.error(msg)
             raise Exception(msg)
 
         # Read MaxMind license key
         license_key = self.get_max_mind_license_key()
         if not license_key:
-            msg = "Max Mind license key not found in password store. Please update config from Max Mind database configuration page.."
+            msg = "Max Mind license key not found in password store. Please update config from Max Mind database configuration page."
             logger.error(msg)
             raise Exception(msg)
 
+        # Read Proxy URL
+        proxy_url = None
+        try:
+            proxy_url = self.get_max_mind_proxy_url()
```

Comment:
Have you tested with the proxy password having special characters in it? if it has @ then it should be passed in encoded format.

Reply:
Thanks for catching this. 

I have not tested it with real proxy but I have tested the functions with below combinations

```
print(encode_username_password_in_proxy_url("https://1.1.1.1"))
print(encode_username_password_in_proxy_url("https://1.1.1.1:8888"))
print(encode_username_password_in_proxy_url("https://abc:xyz@1.1.1.1:8888"))
print(encode_username_password_in_proxy_url("https://abc:hy%rsz*n@1.1.1.1:8888"))
print(encode_username_password_in_proxy_url("https://sbc:vatsal@123@1.1.1.1:8888"))
print(encode_username_password_in_proxy_url("https://vatsal@crossrealms.com:vatsal@123@1.1.1.1:8888"))
```

Answers

```
https://1.1.1.1
https://1.1.1.1:8888
https://abc:xyz@1.1.1.1:8888
https://abc:hy%25rsz%2An@1.1.1.1:8888
https://sbc:vatsal%40123@1.1.1.1:8888
https://vatsal%40crossrealms.com:vatsal%40123@1.1.1.1:8888
```

Reply:
Just check that urlparse supports the socks4 and socks5 scheme extraction. rest changes are good.

Reply:
As per the doc, it supports only these schemes: https://docs.python.org/3/library/urllib.parse.html
![image](https://github.com/CrossRealms/Splunk-App-Auto-Update-MaxMind-Database/assets/57986609/4bd74d90-ef72-4619-94ee-009c3e7b74e6)

Reply:
Will look into it and try to solve it some other way.

Reply:
Assuming user won't have colon (:) in it name itself.  Rest of all cases are covered and looks good.